pysat/_instrument.py

```diff
@@ -2606,349 +2084,881 @@ def next(self, verifyPad=False):
+        self.download(date_array=new_dates, **kwargs)
+
+    def download(self, start=None, stop=None, freq='D', date_array=None,
+                 **kwargs):
+        """Download data for given Instrument object from start to stop.
 
         Parameters
         ----------
-        input_meta : Meta
-            The metadata object to translate
+        start : pandas.datetime (yesterday)
+            start date to download data
+        stop : pandas.datetime (tomorrow)
+            stop date (inclusive) to download data
+        freq : string
+            Stepsize between dates for season, 'D' for daily, 'M' monthly
+            (see pandas)
+        date_array : list-like
+            Sequence of dates to download date for. Takes precedence over
+            start and stop inputs
+        **kwargs : dict
+            Dictionary of keywords that may be options for specific instruments.
+            The keyword arguments 'user' and 'password' are expected for remote
+            databases requiring sign in or registration.
 
-        Returns
-        -------
-        dict
-            A dictionary of the metadata for each variable of an output file
-            e.g. netcdf4
+        Note
+        ----
+        Data will be downloaded to pysat_data_dir/patform/name/tag
+
+        If Instrument bounds are set to defaults they are updated
+        after files are downloaded.
 
         """
-        export_dict = {}
-        if self._meta_translation_table is not None:
-            # Create a translation table for the actual values of the meta
-            # labels. The instrument specific translation table only stores the
-            # names of the attributes that hold the various meta labels
-            translation_table = {}
-            for key in self._meta_translation_table:
-                translation_table[getattr(self, key)] = \
-                    self._meta_translation_table[key]
-        else:
-            translation_table = None
-        # First Order Data
-        for key in input_meta.data.index:
-            if translation_table is None:
-                export_dict[key] = input_meta.data.loc[key].to_dict()
-            else:
-                # Translate each key if a translation is provided
-                export_dict[key] = {}
-                meta_dict = input_meta.data.loc[key].to_dict()
-                for orig_key in meta_dict:
-                    if orig_key in translation_table:
-                        for translated_key in translation_table[orig_key]:
-                            export_dict[key][translated_key] = \
-                                meta_dict[orig_key]
-                    else:
-                        export_dict[key][orig_key] = meta_dict[orig_key]
+        # make sure directories are there, otherwise create them
+        try:
+            os.makedirs(self.files.data_path)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
```

Comment:
Need a descriptive variable name instead of `e` and a descriptive raise type and message.